test/integration/metrics_test.go

```diff
@@ -11,15 +12,102 @@ import (
 	"testing"
 	"time"
 
-	"github.com/newrelic/nri-docker/src/biz"
-
 	"github.com/docker/docker/client"
+	"github.com/newrelic/infra-integrations-sdk/log"
 	"github.com/newrelic/infra-integrations-sdk/persist"
-	"github.com/newrelic/nri-docker/src/raw"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/newrelic/nri-docker/src/biz"
+	"github.com/newrelic/nri-docker/src/raw"
+	"github.com/newrelic/nri-docker/src/raw/dockerapi"
 )
 
+func TestCompareMetrics(t *testing.T) {
+	// GIVEN a container consuming a lot of CPU
+	containerID, dockerRM := stress(t, "stress-ng", "-c", "2", "-l", "50", "-t", "5m", "--vm", "1", "--vm-bytes", "140M", "--iomix", "10")
+	defer dockerRM()
+
+	// WHEN its metrics are sampled and processed from CGroups
+	dockerClient := newDocker(t)
+	defer dockerClient.Close()
+
+	cgroupFetcher := fetcher(t, dockerClient)
+	metricsCGroup := biz.NewProcessor(
+		persist.NewInMemoryStore(),
+		cgroupFetcher,
+		dockerClient,
+		0)
+	sampleCGroup, err := metricsCGroup.Process(containerID)
+	require.NoError(t, err)
+
+	// WHEN its metrics are sampled and processed from API Server
+	info, err := dockerClient.Info(context.Background())
+	require.NoError(t, err)
+	if info.CgroupVersion != "2" {
+		t.Skip("DockerAPIFetcher only supports cgroups v2 version")
+	}
+
+	fetcherAPI := dockerapi.NewFetcher(dockerClient)
+	metricsAPI := biz.NewProcessor(
+		persist.NewInMemoryStore(),
+		fetcherAPI,
+		dockerClient,
+		0)
+	sampleAPI, err := metricsAPI.Process(containerID)
+	require.NoError(t, err)
+
+	// Core metrics are calculated from metrics.Process time differences, using variables with seconds accuaracy. Use a tick larger than a second for accuracy.
+	time.Sleep(10 * time.Second)
+	assert.EventuallyWithT(t,
+		func(t *assert.CollectT) {
+			sampleAPI, err = metricsAPI.Process(containerID)
+			require.NoError(t, err)
+			data, _ := json.Marshal(sampleAPI)
+			log.Error("sampleAPI: %q", string(data))
+
+			sampleCGroup, err = metricsCGroup.Process(containerID)
+			require.NoError(t, err)
+			data, _ = json.Marshal(sampleCGroup)
+			log.Error("sampleCGroup: %q", string(data))
+
+			// TODO this comparisons should be enabled back as soon as they are fixed for cgroupsV2
+			// assert.InDelta(t, sampleCGroup.Memory.SoftLimitBytes, sampleAPI.Memory.SoftLimitBytes, 50000, "SoftLimitBytes")
+			// assert.InDelta(t, sampleCGroup.CPU.Shares, sampleAPI.CPU.Shares, 200, "Shares")
+
+			// These metrics cannot be tested since in the Github runner the metrics are not reported
+			// assert.InDelta(t, sampleCGroup.BlkIO.TotalWriteCount, sampleAPI.BlkIO.TotalWriteCount, 5000, "TotalWriteCount")
+			// assert.InDelta(t, sampleCGroup.BlkIO.TotalReadCount, sampleAPI.BlkIO.TotalReadCount, 500, "TotalReadCount")
+
+			// These metric is not available for fargate and through he DockerAPI
```

Comment:
```suggestion
			// These metrics are not available for fargate and through the DockerAPI
```

I'd add the 2 @sigilioso [mentioned](https://github.com/newrelic/nri-docker/pull/220/files#r1502256593) in this section.